proposals/tf_utf8_identifiers/README.md

```diff
@@ -0,0 +1,443 @@
+
+The proposed implementation adds four compile time initialized maps that map:
+
+- Single code point lowercase to uppercase
+- Single code pointer uppercase to lowercase
+- Single code point lowercase to multiple code point uppercase
+- Single code point uppercase to multiple code point lowercase
+
+All values in these maps are 32-bit integers representing the codepoint of the Unicode character in question (derived from the UTF-8 byte sequence for the character).  Appropriate modifications were made to the following `Tf` methods:
+- `TfStringToLower`
+- `TfStringToUpper`
+- `TfStringCapitalize`
+
+Different maps were used so that single code point maps (the most common) would have have to be stored as vectors of length = 1 (at the cost of two lookups, one to determine if the source codepoint is contained in the single codepoint map and, if not, another to determine if its in the multi-codepoint map).  If a codepoint is not present in one of the maps, there is no case mapping for that character (i.e., it maps to itself).
+
+The proposed implementation for Unicode strings is not exhaustive.  There are a number of operations in the `Tf` library that would likely benefit from having specific Unicode implementations, but which are not provided by the proposed implementation.  An investigation of these operations yielded at least the following groups of methods:
+- Capturing Unicode characters representing numeric values not in the traditional ASCII value sequence
+  - `TfStringToLong`
+  - `TfStringToULong`
+  - `TfStringToInt64`
+  - `TfStringToUInt64`
+  - `TfStringToDouble`
+- Tokenization methods that rely on delimiters which are assumed to fit into a single byte character (though they are mostly used to tokenize things separated by `.` or `/` characters for e.g., paths, a robust implementation would assume any multi-byte character or sequence of characters as the delimiter)
+  - `TfStringTrimLeft`
+  - `TfStringTrimRight`
+  - `TfStringTrim`
+  - `TfStringTokenize`
+  - `TfQuotedStringTokenize`
+  - `TfStringSplit`
+  - `TfStringTokenizeToSet`
+  - `TfStringJoin`
+- Substring based methods that are not robust under case folding
+  - `TfStringContains`
+  - `TfStringStartsWith`
+  - `TfStringEndsWith`
+- String concatenation - Unicode normalization forms are not closed under string concatenation; specifically at least the area where the strings are joined need to be considered for composition
+  - `TfStringJoin`
+
+It may be interesting to consider the impact of C++ 20's `std::u8string` type for the methods listed above.  Overloads of these methods that take a `std::u8string` parameter would provide very clear semantics to a developer about the treatment of string content.  While the use of `std:u8string` would really only affect the copmiler encoding of string literals, the burden is then placed on the developer to choose explicitly which of the methods to use.
```

Comment:
copmiler  => compiler

Reply:
Proofreading is hard :/  fixed! (and thanks!)